src/libraries/System.Net.Http/src/System/Net/Http/HttpRequestMessage.cs

```diff
@@ -183,7 +185,28 @@ private void InitializeValues(HttpMethod method, Uri? requestUri)
 
         internal bool MarkAsSent()
         {
-            return Interlocked.Exchange(ref _sendStatus, MessageAlreadySent) == MessageNotYetSent;
+            return Interlocked.CompareExchange(ref _sendStatus, MessageAlreadySent, MessageNotYetSent) == MessageNotYetSent;
+        }
+
+        internal void MarkAsTrackedByTelemetry()
+        {
+            Debug.Assert(_sendStatus != MessageShouldEmitTelemetry);
+            _sendStatus = MessageShouldEmitTelemetry;
```

Comment:
Isn't this going to overwrite the MessageAlreadySent set as part of MarkAsSent?

Reply:
Oh, I see, you consider both MessageAlreadySent and MessageShouldEmitTelemetry to mean already sent.

Reply:
Would

```diff
private const int MessageNotYetSent = 0;
private const int MessageAlreadySent = 1;
-private const int MessageShouldEmitTelemetry = 2;
+private const int MessageAlreadySent_StopNotYetCalled = 2;
```

be more clear?